test/stress/pool-create-hanging.js

```diff
@@ -0,0 +1,193 @@
+    setTimeout(resolve, ms)
+  })
+}
+
+function cloneDeep (val) {
+  return JSON.parse(JSON.stringify(val))
+}
+
+// initial dummy object, which will be overridden by connection process
+let mssqlCon = { connected: false }
+let connectionNumber = 0
+
+async function mssqlQuery (sql) {
+  // recreate connection is not connected / connecting stus
+  if (!mssqlCon.connected && !mssqlCon.connecting) {
+    debug('========== Reconnecting mssql')
+
+    // this test needs toxiproxy to run so docker mssql is also required
+    mssqlCon = new mssql.ConnectionPool({
+      port: 21433,
+      user: 'sa',
+      password: 'S0meVeryHardPassword',
+      server: 'localhost',
+      requestTimeout: 5000,
+      connectionTimeout: 7000,
+      debug: true,
+      pool: {
+        debug: true,
+        min: 1,
+        max: 1,
+        acquireTimeoutMillis: 10000,
+        testOnBorrow: true
+      }
+    })
+
+    mssqlCon.connectionNumber = connectionNumber += 1
+
+    // NOTE: this never seems to fire
+    mssqlCon.on('error', err => {
+      debug('There was fatal connection error %O', err)
+    })
+
+    debug('Waiting for connection')
+    await mssqlCon.connect()
+  }
+
+  debug('>>>>>>>>>>>>>>>> Running query >>>>>>>>>>>>>>>')
+  debug('%o', {
+    peding: mssqlCon.pool.pending,
+    size: mssqlCon.pool.size,
+    available: mssqlCon.pool.available,
+    spareResourceCapacity: mssqlCon.pool.spareResourceCapacity,
+    borrowed: mssqlCon.pool.borrowed
+  })
+  const res = await mssqlCon.request().query(sql)
+  debug('query response %o', res.recordset)
+  debug('<<<<<<<<<<<<<<<<< query  done <<<<<<<<<<<<<<<<')
+  return res
+}
+
+// setup counters to check queries / results / errors made since last report
+const counters = {}
+function setQueryCounters (name) {
+  counters[name] = {queries: 0, results: 0, errors: 0}
+}
+setQueryCounters('mssql')
+
+// start printing out counters
+let lastCounters = cloneDeep(counters)
+let hangRounds = 0
+let statsInterval = setInterval(() => {
+  const reqsPerSec = {}
+  for (let key of Object.keys(counters)) {
+    reqsPerSec[key] = {
+      queries: (counters[key].queries - lastCounters[key].queries),
+      results: (counters[key].results - lastCounters[key].results),
+      errors: (counters[key].errors - lastCounters[key].errors)
+    }
+  }
+
+  console.log('------------------------ REQUESTS SINCE LAST PRINT ------------------------')
+  if (mssqlCon.pool) {
+    console.dir({
+      peding: mssqlCon.pool.pending,
+      size: mssqlCon.pool.size,
+      available: mssqlCon.pool.available,
+      borrowed: mssqlCon.pool.borrowed
+    })
+  }
+  console.dir(reqsPerSec, { colors: true })
+  console.log('------------------------------------ EOS ---------------------------------')
+  lastCounters = cloneDeep(counters)
+
+  if (reqsPerSec.mssql.queries === 0) {
+    hangRounds++
+  } else {
+    hangRounds = 0
+  }
+
+  if (hangRounds === 3) {
+    console.log('Driver did hang exit with error')
+    process.exit(1)
```

Comment:
since this is a test, does it make sense to exit process and not just fail an assertion?

Reply:
Probably throwing exception and leaking it out kill the process would work too, but I think that calling exit(1) reflects more explicitly about that it is really intended to close this this process if hang is detected.